aoc-client/src/lib.rs

```diff
@@ -421,6 +421,79 @@ impl AocClient {
         Ok(())
     }
 
+    fn get_event_stats_html(&self) -> AocResult<String> {
+        debug!("🦌 Fetching event stats for {}", self.year);
+        let url = format!("https://adventofcode.com/{}/stats", self.year);
+
+        let response = http_client(&self.session_cookie, "text/html")?
+            .get(url)
+            .send()?;
+        let contents = response.error_for_status()?.text()?;
+
+        let main = Regex::new(r"(?i)(?s)<main>(?P<main>.*)</main>")
+            .unwrap()
+            .captures(&contents)
+            .ok_or(AocError::AocResponseError)?
+            .name("main")
+            .unwrap()
+            .as_str()
+            .to_string();
+
+        Ok(main)
+    }
+
+    pub fn show_event_stats(&self) -> AocResult<()> {
+        let stats_html = self.get_event_stats_html()?;
+
+        let stats_text = from_read_with_decorator(
+            stats_html.as_bytes(),
+            self.output_width,
+            TrivialDecorator::new(),
+        );
+
+        // number of completions loosely associated with each star
+        let star_val = stats_text
+            .split_whitespace()
+            .find(|chunk| chunk.parse::<u64>().is_ok())
+            .unwrap()
```

Comment:
This unwrap is a bit dangerous. We need to handle it in case the page format changes in the future (we could return a AocResponseError to flag this). For instance, this currently panics if you ask for the 2023 stats, which gets an almost blank page with a "no stats yet" message.

Reply:
That makes sense. I initially avoided using a regex because I don't have any experience using them (yet), but I'll look into this. Thanks!